src/main/java/com/sourcegraph/webhook/registry/WebhookRegistry.java

```diff
@@ -79,23 +78,26 @@ public static void register(Webhook hook) throws WebhookException {
         String scope = resolveScopeID(hook.scope);
 
         activeObjects.executeInTransaction(() -> {
-            Map<String, Object> params = new HashMap<>();
-            params.put("NAME", hook.name);
-            params.put("SCOPE", scope);
-            params.put("ENDPOINT", hook.endpoint);
-            params.put("SECRET", hook.secret);
-
-            WebhookEntity hookEntity = activeObjects.create(WebhookEntity.class, params);
-            hookEntity.save();
+            WebhookEntity[] ents = activeObjects.find(WebhookEntity.class, Query.select().where("ID = ?", hook.id));
+            WebhookEntity ent = ents.length == 0 ? activeObjects.create(WebhookEntity.class) : ents[0];
+            ent.setName(hook.name);
+            ent.setScope(scope);
+            ent.setScope(hook.endpoint);
+            ent.setScope(hook.secret);
+            ent.save();
+
+            for (EventEntity event : ent.getEvents()) {
+                activeObjects.delete(event);
```

Comment:
Why are we deleting events and re-creating them?

Reply:
I changed the code to remove old events and create new ones instead of destroying/recreating all.

Reply:
I don't understand why this is necessary. Can you explain?

Reply:
This is to update (upsert) registered events for a webhook. For example: if a webhook is registered for `pr:comment:added` + `pr:comment:deleted` and we want to update it to `pr:reviewers:updated`, this logic would remove the older events and add in the desired one. 

I realize most of this logic won't be (isn't) used at all. I plan to overhaul/remove it in 3.13.

Reply:
> his logic would remove the older events and add in the desired one.

So, are events stored in the database forever regardless of registered webhooks? I don't understand why there needs to be any event stored — we should be only sending it out if there's a registered webhook that matches.

> I realize most of this logic won't be (isn't) used at all for automation. I plan to overhaul/remove it in 3.13. It is here purely for a complete REST implementation.

How exactly is it not used? Please elaborate and be explicit about what work you're proposing to do in 3.13.